frontend/src/components/MapView/LeftPanel/utils.ts

```diff
@@ -19,20 +20,26 @@ type LayersCategoriesType = LayersCategoryType[];
 
 type MenuItemsType = MenuItemType[];
 
-function formatLayersCategories(layersList: {
-  [key: string]: Array<LayerKey | TableKey>;
-}): LayersCategoriesType {
+function formatLayersCategories(
+  layersList: {
+    [key: string]: Array<LayerKey | TableKey>;
+  },
+  selectedLayers?: LayerType[],
+): LayersCategoriesType {
   return map(layersList, (layerKeys, layersListKey) => ({
     title: startCase(layersListKey),
     layers: layerKeys.filter(isLayerKey).map(key => {
       if (typeof key === 'object') {
         const group = mapKeys(key, (_v, k: string) =>
           camelCase(k),
         ) as unknown as MenuGroup;
-        const mainLayer = group.layers.find(l => l.main);
-        const layer = LayerDefinitions[mainLayer?.id as LayerKey];
-        // eslint-disable-next-line fp/no-mutation
-        layer.group = group;
+        const visibleLayer =
+          group.layers.find(gl =>
+            selectedLayers?.some(sl => sl.id === gl.id),
+          ) || group.layers.find(l => l.main);
+
+        const staticLayer = LayerDefinitions[visibleLayer?.id as LayerKey];
+        const layer = { ...staticLayer, group };
```

Comment:
@gislawill I think that's where the issue comes from for grouped layers. We end up with multiple layers in the timeline when we only want the `main` one.

Can you explain this change? Could we just do `const layer = staticLayer`

Reply:
This change supported allow the same left panel menu item to support two different layers (in this case, monthly vs seasonal). I see the issue you're referring to but I don't think it stems for this change — the issue is in the timeline view and this change is specifically for the menu items. 

I'm going to investigate what updates were made in the timeline view involving grouped layers.

Reply:
@ericboucher you were right that this function was the source of the error. By using `const layer = { ...staticLayer, group };` instead of `layer.group = group`, I was avoiding the eslint error (fp/no-mutation) without realizing that modifying the layer definition in our global constants was exactly what we were trying to do. 

Because I updated this formatLayersCategories function to run during renders (getDynamicMenuList) rather than just at start up, a new error occurred when writing to our layer definition (they're frozen sometime after initialization but before renders it seems). I put in place a fix for that so everything functions as expected and added comments explaining the fix.

I should note, I don't think that writing to our global layer constants (LayerDefinitions) from within a menu formatter is a great pattern to follow. Rather, we should write the group property when creating the LayerDefinitions. Because this both a pre-existing issue and this PR is already way too large, I'd prefer to that fix that separately. I'd be happy to create an issue if you agree.

Reply:
Agreed